File: src/main/kotlin/data/RPData.kt
package ua.pp.lumivoid.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

//Not all data here, because it seems useless and I don't want to manually add it
// https://api.radioplayer.ua/config

@Serializable
data class RPData(
    val stations: List<StationData>,
)

@Serializable
data class StationData(
    val id: Int,
    val enabled: Boolean,
    @SerialName("sorting_id") val sortingId: Int,
    val url: String,
//    @SerialName("filtering_tags") val filteringTags: List<String>,
    val names: Map<String, String>,
    val tags: Map<String, String>,
    @SerialName("logo") val logoUrl: String,
//    @SerialName("playlist") val playlistUrl: String,
    @SerialName("current_song") val currentSongUrl: String,
    val streams: StreamsData,
)

@Serializable
data class StreamsData(
    val regular: Map<String, String>,
    val hd: Map<String, String>,
)

File: Src/Data/RpData.cs
using System.Text.Json.Serialization;
using System.Collections.Generic;

namespace RadioPlayerForDesktop.Data;

public record RpData([property: JsonPropertyName("stations")] List<RpData.StationData> Stations)
{
    public record StationData(
        [property: JsonPropertyName("id")] int Id,
        [property: JsonPropertyName("enabled")] bool Enabled,
        // [property: JsonPropertyName("sorting_id")] int SortingId,
        // [property: JsonPropertyName("url")] string Url,
        [property: JsonPropertyName("names")] Dictionary<string, string> Names,
        // [property: JsonPropertyName("tags")] Dictionary<string, string> Tags,
        [property: JsonPropertyName("logo")] string LogoUrl,
        // [property: JsonPropertyName("current_song")] string CurrentSongUrl,
        [property: JsonPropertyName("streams")] StationData.StreamsData Streams
    )
    {

        public record StreamsData(
            [property: JsonPropertyName("regular")] Dictionary<string, string> Regular,
            [property: JsonPropertyName("hd")] Dictionary<string, string> Hd
        );
    }
    
}